### ISSUE TITLE
Add a "Terminology" (or similar) section to the template

### ISSUE THREAD

**Issue by nicopap** [bevyengine/rfcs#61]
A terminology section would define jargon specific to an RFC.

A terminology helps RFC editing and reading:
- Helps the author re-use the same terms to refer to identical concepts
- Helps the author avoid mixing different ideas, on surface identical but not interchangeable
- Helps the author identify the concepts they are using and structure their explanation with clearly defined concepts in mind
- Helps the author limit how many different concepts they handle in the same RFC
- The reader can refer to that section if they forgot the meaning of a particularly specialized term
- The reader can pay closer attention to occurrence of terms specified in the terminology section
- It makes RFCs more accessible, as it introduces context and jargon that non-experts may not know

This is different from the "User-facing explanation" concept sub-section, in that the terminology section is for the terms used in the RFC rather than just the ones introduced into the user-facing API.

Alternative names for this section are
- Terminology
- Jargon
- syllabus
- nomenclature

**Comment by alice-i-cecile** [comment]
Strongly in favor of this. I think that my favorite terms are: "Definitions", "Glossary" and "Terminology".

**Comment by djeedai** [comment]
Agreed, I started writing a RFC and this is the first thing I added. I vote for Terminology or Glossary.
